rust/cardano-blockchain-types/src/metadata/cip36/key_registration.rs

```diff
@@ -76,6 +76,7 @@ enum Cip36KeyRegistrationKeys {
 }
 
 impl Decode<'_, ()> for Cip36KeyRegistration {
+    #[allow(clippy::too_many_lines)]
     fn decode(d: &mut Decoder, ctx: &mut ()) -> Result<Self, decode::Error> {
```

Comment:
Suggest moving the problem report into the context, and only using decode:Error if the decode actually fails.
This way, if you can decode, but its internally problematic, we have what we could decode and the report, but if the data is invalid (say its just not proper cbor) then we fail when we hit that, with an error, but the report still captures everything that failed before then.